crates/bevy_ecs/src/entity/visit_entities.rs

```diff
@@ -1,48 +1,75 @@
+use std::iter;
+
 use crate::{
     entity::Entity,
     identifier::masks::{IdentifierMask, HIGH_MASK},
     world::World,
 };
+pub use bevy_ecs_macros::IterEntities;
 
 use super::EntityHashMap;
 
-/// Operation to map all contained [`Entity`] fields in a type to new values.
-///
-/// As entity IDs are valid only for the [`World`] they're sourced from, using [`Entity`]
-/// as references in components copied from another world will be invalid. This trait
-/// allows defining custom mappings for these references via [`EntityMappers`](EntityMapper), which
-/// inject the entity mapping strategy between your `MapEntities` type and the current world
-/// (usually by using an [`EntityHashMap<Entity>`] between source entities and entities in the
-/// current world).
+/// Apply an operation to all entities in a container.
 ///
-/// Implementing this trait correctly is required for properly loading components
-/// with entity references from scenes.
-///
-/// ## Example
+/// This is implemented by default for types that implement [`IterEntities`] or
+/// where `&T` and `&mut T` both implement [`IntoIterator`].
```

Comment:
I think it's obvious from the docs? You can see blanked impls, not sure if it worth to repeat information like this.

Reply:
I think it's nice to have this context located close by the next bit that explains situations in which you'd need to implement it directly.